xapian-bindings/java/java.i

```diff
@@ -174,21 +174,63 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%ignore MSetIterator::next();
+%javamethodmodifiers MSetIterator::next_helper() "private"
+
 %extend MSetIterator {
-    Xapian::docid next () {
-	Xapian::docid tmp;
+    void next_helper () {
 	if (Xapian::iterator_valid(*self)) {
```

Comment:
Indeed - we can't just inline one in the other since they're written in different languages (`next()` is in Java, `next_helper()` is in C++).